ibis/backends/tests/test_register.py

```diff
@@ -459,18 +461,17 @@ def test_read_parquet(con, tmp_path, data_dir, fname, in_table_name, out_table_n
         "mysql",
         "pandas",
         "postgres",
-        "snowflake",
         "sqlite",
         "trino",
     ]
 )
-def test_read_csv(con, data_dir, fname, in_table_name, out_table_name):
+def test_read_csv(con, data_dir, in_table_name, out_table_name, num_diamonds):
+    fname = "diamonds.csv"
     with pushd(data_dir / "csv"):
         if con.name == "pyspark":
             # pyspark doesn't respect CWD
             fname = str(Path(fname).absolute())
         table = con.read_csv(fname, table_name=in_table_name)
 
     assert any(out_table_name in t for t in con.list_tables())
-    if con.name != "datafusion":
-        table.count().execute()
+    assert table.count().execute() == num_diamonds
```

Comment:
Could we also test that e.g. the first 10 rows match with the expected data?

Asserting only on `count()` can hide issues with the schema.

Reply:
I'd rather not, the ordering may not be consistent across backends. Happy to check the schema though.

Reply:
Added an assertion about the `schema`

Reply:
Thanks!